core/coreunix/add.go

```diff
@@ -90,49 +150,176 @@ func AddWrapped(n *core.IpfsNode, r io.Reader, filename string) (string, *merkle
 	return gopath.Join(k.String(), filename), dagnode, nil
 }
 
-func add(n *core.IpfsNode, reader io.Reader) (*merkledag.Node, error) {
-	mp := n.Pinning.GetManual()
+func NewAdder(ctx context.Context, n *core.IpfsNode, out chan interface{}, progress bool, hidden bool, trickle bool, wrap bool, chunker string) Adder {
```

Comment:
maybe instead of such a big function, can have an init like:

```go
a := NewAdder(ctx, n, out)
a.Progress = true // defaults to false
a.Hidden = false // defaults to true
a.Trickle = true // defaults to false
a.Chunker = "rabin" // defaults to ""
```